### PR TITLE
Supply custom build scripts for CodeQL

### PR DESCRIPTION
**Pull request by friendlyanon**
CodeQL will match the entire repository for specific filenames to use as build scripts, so putting these in a subdirectory is fine.

Fixes https://github.com/microsoft/ifc/issues/29#issuecomment-1751866508

### CODE REVIEW

File: codeql/build.bat [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+:: Build file for CodeQL
+
+cd ..
+
+cmake --preset=ci-windows ^
+-D CMAKE_TOOLCHAIN_FILE= ^
+-D "CMAKE_MODULE_PATH:PATH=%cd%\cmake\find"
+if not %errorlevel%==0 exit /b %errorlevel%
+
+cmake --build build --config Debug
```
Comment (friendlyanon) [inline codeql/build.bat:12]: Is debug the desired build config for CodeQL runs?

Comment (GabrielDosReis) [reply codeql/build.bat:12]: We want the `Release` config for CodeQL

**Comment by GabrielDosReis** [comment]
> CodeQL will match the entire repository for specific filenames to use as build scripts, so putting these in a subdirectory is fine.
> 
> Fixes [#29 (comment)](https://github.com/microsoft/ifc/issues/29#issuecomment-1751866508)

Thanks for looking into this; very much appreciated!
Could you rework this PR based on the conversion of the implicit CodeQL build into the exploit workflow in #41 ?

**Comment by friendlyanon** [comment]
Shouldn't CodeQL just be part of the normal CI workflow? `cmake.yml` only needs 2 new steps and that'd be that.

**Comment by GabrielDosReis** [comment]
> Shouldn't CodeQL just be part of the normal CI workflow? 

Yes, it is.  Until now, it had been implicit, and the results shown only after check-in is performed.  Now, we are making it explicitly part of the normal CI workflow.

**Review by GabrielDosReis — APPROVED** [review]

**Pull request merged**